tools/module_dependencies.py

```diff
@@ -0,0 +1,149 @@
+__all__ = ['modules_dependent_on', 'dependency_graph']
+
+
+import numpy as np
+import importlib
+import multiprocessing
+import functools
+import sys
+import os
+import glob
+import pytest
+
+
+package = 'skimage'
+included_mods = [
+    '_shared',
+]
+excluded_mods = [
+    '__version__',
+]
+
+
+@functools.cache
+def _pkg_modules() -> tuple[list[str], dict[str, int]]:
+    """List all package submodules.
+
+    Returns
+    -------
+    submodules : list
+        Sorted list of package submodules.
+    submodule_idx : dict
+        Mapping of submodule to integer, its index
+        in `submodules`.
+    """
+    pkg = importlib.import_module(package)
+
+    members = {f'{package}.{attr}' for attr in dir(pkg)}
+    included_mods_full = {f'{package}.{mod}' for mod in included_mods}
+    excluded_mods_full = {f'{package}.{mod}' for mod in excluded_mods}
+
+    submodules = sorted((members | included_mods_full) - excluded_mods_full)
+
+    # Sort entries, so that all adjacency matrix calculations are stable
+    submodule_idx = {mod: index for index, mod in enumerate(submodules)}
+
+    # # Clear out sys.modules for when we spawn our import detector
+    pkg_mods = [mod for mod in sys.modules if f"{package}." in mod]
+    for mod in pkg_mods:
+        del sys.modules[mod]
+
+    return submodules, submodule_idx
+
+
+def _import_dependencies(module: str) -> set[str]:
+    bad_sys_modules = {mod for mod in sys.modules if f"{package}." in mod}
+    if bad_sys_modules:
+        raise RuntimeError(
+            f"Yikes! No package modules present at this point, but seeing {bad_sys_modules}!"
+        )
+
+    importlib.import_module(module)
+
+    # This is a workaround for identifying the test modules
+    # associated with an skimage submodule.
+    #
+    # Other libraries would need to do something different, and
+    # we should think how to generalize.
```

Comment:
~~Is this because you don't have any sub-sub-packages that have their own tests?~~ It looks like you do. WIll this catch `filters.rank.tests`?